code/modules/xenoarchaeology/traits/xenoartifact_majors.dm

```diff
@@ -0,0 +1,589 @@
+///============
+/datum/xenoartifact_trait/major/laser
+	desc = "Barreled"
+	label_desc = "Barreled: The shape resembles the barrel of a gun. It's possible that it might dispense candy."
+	flags = PLASMA_TRAIT | URANIUM_TRAIT
+
+/datum/xenoartifact_trait/major/laser/activate(obj/item/xenoartifact/X, atom/target, mob/living/user)
+	if(isliving(target) && get_dist(target, X.loc || user) <= 1)
+		var/mob/living/victim = target
+		victim.adjust_fire_stacks(5*(X.charge/X.charge_req))
+		victim.IgniteMob()
+		return
+	var/obj/item/projectile/A
+	switch(X.charge)
+		if(0 to 24)
+			A = new /obj/item/projectile/beam/disabler
+		if(25 to 79)
+			A = new /obj/item/projectile/beam/laser
+		if(80 to 200)
+			A = new /obj/item/ammo_casing/energy/laser/heavy
+	A.preparePixelProjectile(get_turf(target), X)
+	A.fire()
+	playsound(get_turf(src), 'sound/mecha/mech_shield_deflect.ogg', 50, TRUE) 
+
+///============
+/// Corginator, turns the target into a corgi for a short time
+///============
+/datum/xenoartifact_trait/major/corginator ///All of this is stolen from corgium. 
+	desc = "Fuzzy" //Weirdchamp
+	label_desc = "Fuzzy: The shape is hard to discern under all the hair sprouting out from the surface. You swear you've heard it bark before."
+	flags = BLUESPACE_TRAIT
+	///List of all affected targets, used for early qdel
+	var/list/victims = list()
+	///list of targets by corgi
+	var/list/target_keys = list()
+	///list of timers for transform back
+	var/list/timers = list()
+
+/datum/xenoartifact_trait/major/corginator/activate(obj/item/xenoartifact/X, mob/living/target)
+	X.say(pick("Woof!", "Bark!", "Yap!"))
+	if(!(SSzclear.get_free_z_level()))
+		playsound(get_turf(X), 'sound/machines/buzz-sigh.ogg', 50, TRUE)
+		return
+	if(istype(target, /mob/living) && !(istype(target, /mob/living/simple_animal/pet/dog/corgi)))
+		var/mob/living/simple_animal/pet/dog/corgi/new_corgi = transform(target)
+		timers[new_corgi] = addtimer(CALLBACK(src, .proc/transform_back, new_corgi), (X.charge*0.6) SECONDS, TIMER_STOPPABLE)
+		victims |= new_corgi
+		target_keys[new_corgi] = target
+		X.cooldownmod = (X.charge*0.8) SECONDS
+
+/datum/xenoartifact_trait/major/corginator/proc/transform(mob/living/target)
+	if(!istype(target))
+		return
+	var/obj/shapeshift_holder/H = locate() in target
+	if(H)
+		playsound(get_turf(target), 'sound/machines/buzz-sigh.ogg', 50, TRUE)
+		return
+	ADD_TRAIT(target, TRAIT_NOBREATH, CORGIUM_TRAIT)
+	var/mob/living/simple_animal/pet/dog/corgi/new_corgi = new(target.loc)
+	H = new(new_corgi,src,target)
+	//hat check
+	var/mob/living/carbon/C = target
+	if(istype(C))
+		var/obj/item/hat = C.get_item_by_slot(ITEM_SLOT_HEAD)
+		if(hat?.dog_fashion)
+			new_corgi.place_on_head(hat,null,FALSE)
+	RegisterSignal(new_corgi, COMSIG_MOB_DEATH, .proc/transform_back)
+	return new_corgi
+
+/datum/xenoartifact_trait/major/corginator/proc/transform_back(mob/living/simple_animal/pet/dog/corgi/new_corgi)
+	var/mob/living/target = (target_keys[new_corgi])
```

Comment:
Why do we need this? Can't we just send it through in the parameters of the timer.
Or you could just locate the shapeshift holder, since I'm pretty sure that contains a reference to the original mob

Reply:
Locating SS holder